File: packages/dotnet-sdk/src/Blindfold.Sdk/Errors/BlindfoldException.cs
namespace Blindfold.Sdk.Errors;

public class BlindfoldException : Exception
{
    public int? StatusCode { get; }

    public BlindfoldException(string message) : base(message) { }

    public BlindfoldException(string message, int statusCode) : base(message)
    {
        StatusCode = statusCode;
    }

    public BlindfoldException(string message, Exception innerException) : base(message, innerException) { }
}

File: packages/java-sdk/src/main/java/dev/blindfold/sdk/errors/BlindfoldException.java
package dev.blindfold.sdk.errors;

public class BlindfoldException extends RuntimeException {
    private final int statusCode;

    public BlindfoldException(String message) {
        super(message);
        this.statusCode = 0;
    }

    public BlindfoldException(String message, int statusCode) {
        super(message);
        this.statusCode = statusCode;
    }

    public BlindfoldException(String message, Throwable cause) {
        super(message, cause);
        this.statusCode = 0;
    }

    public int getStatusCode() { return statusCode; }
}
